pkg/config/application.go

```diff
@@ -66,6 +67,48 @@ func NewApplicationGroup(application *Application, mspConfig *mb.MSPConfig) (*cb
 	return applicationGroup, nil
 }
 
+// RemoveAnchorPeer removes an anchor peer from an existing channel config transaction.
+// The removed anchor peer and org it belongs to must both already exist.
+func RemoveAnchorPeer(config *cb.Config, orgName string, anchorPeerToRemove *AnchorPeer) error {
+	if _, orgExists := config.ChannelGroup.Groups["Application"].Groups[orgName]; !orgExists {
+		return fmt.Errorf("application org %s does not exist in channel config", orgName)
+	}
+
+	applicationOrgGroup := config.ChannelGroup.Groups["Application"].Groups[orgName]
+
+	anchorPeersProto := &pb.AnchorPeers{}
+
+	if anchorPeerConfigValue, anchorPeersExist := applicationOrgGroup.Values[AnchorPeersKey]; anchorPeersExist {
+		// Unmarshal existing anchor peers if the config value exists
+		err := proto.Unmarshal(anchorPeerConfigValue.Value, anchorPeersProto)
+		if err != nil {
+			return fmt.Errorf("failed unmarshalling existing anchor peers: %v", err)
+		}
+	}
+
+	existingAnchorPeers := anchorPeersProto.AnchorPeers
+
+	if len(existingAnchorPeers) == 0 {
```

Comment:
I wonder whenever do you really needs this given that if anchor peers of length zero will skip for loop and return error at line 109. Unless you would like to indicate that no anchor peers was configured, when seems that added else clause will cover it.

Reply:
good point, this error is probably not necessary as you mentioned the one on 109 would catch it and I think is meaningful enough to apply in this case as well